plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/bedrock-runtime.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Attributes, DiagLogger, Span, Tracer } from '@opentelemetry/api';
+import { RequestMetadata, ServiceExtension } from './ServiceExtension';
+import {
+  AwsSdkInstrumentationConfig,
+  NormalizedRequest,
+  NormalizedResponse,
+} from '../types';
+
+// Unstable attributes to define inline.
+const SEMATTRS_GEN_AI_SYSTEM = 'gen_ai.system';
+const SEMATTRS_GEN_AI_OPERATION_NAME = 'gen_ai.operation.name';
+const SEMATTRS_GEN_AI_REQUEST_MODEL = 'gen_ai.request.model';
+const SEMATTRS_GEN_AI_REQUEST_MAX_TOKENS = 'gen_ai.request.max_tokens';
+const SEMATTRS_GEN_AI_REQUEST_TEMPERATURE = 'gen_ai.request.temperature';
+const SEMATTRS_GEN_AI_REQUEST_TOP_P = 'gen_ai.request.top_p';
+const SEMATTRS_GEN_AI_REQUEST_STOP_SEQUENCES = 'gen_ai.request.stop_sequences';
+const SEMATTRS_GEN_AI_USAGE_INPUT_TOKENS = 'gen_ai.usage.input_tokens';
+const SEMATTRS_GEN_AI_USAGE_OUTPUT_TOKENS = 'gen_ai.usage.output_tokens';
+const SEMATTRS_GEN_AI_RESPONSE_FINISH_REASONS =
+  'gen_ai.response.finish_reasons';
+
+export class BedrockRuntimeExtension implements ServiceExtension {
```

Comment:
nit: 

```suggestion
export class BedrockRuntimeServiceExtension implements ServiceExtension {
```

The naming pattern from the others seems to be `[ClientName]ServiceExtension`.